### DISCUSSION TITLE
Rebase in Pull Request Etiquette

### DISCUSSION THREAD

**Discussion by Ryuno-Ki** [mdn/mdn-community#550] [Content]
> Does that mean, item 3 in [mdn/content#pull-request-etiquette](https://github.com/mdn/content#pull-request-etiquette) is no longer valid?

I didn’t know we had such a guideline, and so I’ve never followed it. And it runs counter to best practices, so I’d be interested to understand the rationale for it, rather than just following it.

Making additional commits is not a typical or practical way to deal with the common case when rebasing is necessary.

_Originally posted by @sideshowbarker in https://github.com/mdn/content/issues/6460#issuecomment-871582755_

Moving this into a discussion.

@escattone You added this section in https://github.com/mdn/content/commit/71d085c97c18408222e40422e8f2cef720fcae8e / https://github.com/mdn/content/pull/143 to fix https://github.com/mdn/yari/issues/1098 which in turn addresses https://github.com/mdn/yari/issues/1097 - would you be willing to reconsider?

**Comment by escattone** [comment]
@Ryuno-Ki @sideshowbarker Sorry for the delayed reply! All Mozillians were out last week for a wellness week, which was nice!

The guidelines only discourage doing a `git rebase` (rewriting the git history) on a branch after it's been submitted for review in a pull request, simply for the reason that it's more difficult for reviewers to see what's changed since they last reviewed the PR (although I must admit that I'm guilty of doing that from time to time). What @vinyldarkscratch did in https://github.com/mdn/content/pull/6460 (merging `main` into her branch via a merge commit) is totally fine, since she didn't rewrite the git history for her branch.

**Reply by Ryuno-Ki** [reply]
GitHub is patient :-)
No need to excuse.

Yes, a merge occured. I was more irritated by https://github.com/mdn/content/pull/6460#issuecomment-871027120 asking for a rebase.

Personally, I prefer a rebase, as it adds only the changes on top of the most recent status.
Yes, that will screw up the review history. But given that most changes should be small in nature, it shouldn't be that harmful.

I'm fine with following a merge strategy, if that is preferred.

**Reply by escattone** [reply]
@Ryuno-Ki I'd say that if everyone involved in the pull request is OK with rebasing, it's fine. 😄

**Reply by Ryuno-Ki** [reply]
If others signal support, I'd open a PR to update our README.

**Reply by sideshowbarker** [reply]
> If others signal support, I'd open a PR to update our README.

Sounds great to me

**Reply by Ryuno-Ki** [reply]
Okay, finally came around to fill https://github.com/mdn/content/pull/7689 as PR

**Comment by pepelsbey** [comment]
Closing as answered/settled: https://github.com/mdn/content/pull/7689
